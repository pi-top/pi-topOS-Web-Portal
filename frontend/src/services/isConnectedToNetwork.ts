import axios from "axios";

import apiBaseUrl from "./apiBaseUrl";

export default async function isConnectedToNetwork(timeout?: number) {
  const { data } = await axios.get<{connected: boolean}>(
    `${apiBaseUrl}/is-connected`,
    { timeout: timeout || 10000 }
  );

  return data;
}
