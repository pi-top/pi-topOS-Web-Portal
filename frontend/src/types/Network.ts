export type Network = {
  ssid: string;
  passwordRequired: boolean
};

export type NetworkCredentials = {
  ssid: string;
  password: string
};
