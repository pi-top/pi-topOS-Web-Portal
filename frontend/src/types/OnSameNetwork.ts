export type OnSameNetworkData = {
  onSameNetwork: boolean;
  piTopIp: string;
  clientIp: string;
}
