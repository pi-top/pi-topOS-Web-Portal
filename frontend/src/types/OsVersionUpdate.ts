export type OsVersionUpdate = {
  shouldBurn: boolean;
  requireBurn: boolean;
  latestOSVersion: string;
  update: boolean;
};
