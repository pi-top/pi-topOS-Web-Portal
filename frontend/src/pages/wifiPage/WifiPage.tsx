import React, { useEffect, useState } from "react";

import Layout from "../../components/layout/Layout";
import Select from "../../components/atoms/select/Select";

import wifiPageScreen from "../../assets/images/wifi-screen.png";
import refreshIcon from "../../assets/icons/refresh.svg";
import styles from "./WifiPage.module.css";

import Spinner from "../../components/atoms/spinner/Spinner";
import Button from "../../components/atoms/button/Button";
import ConnectDialogContainer from "./connectDialog/ConnectDialogContainer";
import SkipWarningDialog from "./skipWarningDialog/SkipWarningDialog";
import { Network } from "../../types/Network";
import usePrevious from "../../hooks/usePrevious";

export enum ErrorMessage {
  FetchNetworks = "There was a problem getting networks, please refresh the networks list or skip",
}

export enum ExplanationMessage {
  NotConnected = "Choose from the list below.",
  WiredConnection = "Looks like we're already online. You can still connect to a WiFi network.",
  WiFiConnection = "We've connected successfully.",
}

export type Props = {
  onNextClick?: () => void;
  onSkipClick?: () => void;
  onBackClick?: () => void;
  onRefreshClick: () => void;
  networks: Network[];
  isFetchingNetworks: boolean;
  fetchNetworksError: boolean;
  isConnected: boolean;
  connectedNetwork?: Network;
  setConnectedNetwork: (network: Network) => void;
  skipButtonLabel?: string;
  showSkipWarning?: boolean;
};

export default ({
  onNextClick,
  onSkipClick,
  onBackClick,
  onRefreshClick,
  networks,
  isFetchingNetworks,
  fetchNetworksError,
  isConnected,
  connectedNetwork,
  setConnectedNetwork,
  skipButtonLabel,
  showSkipWarning = true,
}: Props) => {
  const previousConnectedNetwork = usePrevious(connectedNetwork);
  const [selectedNetwork, setSelectedNetwork] = useState(connectedNetwork);
  const [isConnectDialogActive, setIsConnectDialogActive] = useState(false);
  const [isSkipWarningDialogActive, setIsSkipWarningDialogActive] =
    useState(false);

  const { ssid: selectedSSID, bssid: selectedBSSID } = selectedNetwork || {};
  const errorMessage = fetchNetworksError && ErrorMessage.FetchNetworks;
  const getExplanation = () => {
    if (connectedNetwork) {
      return ExplanationMessage.WiFiConnection;
    }
    if (isConnected) {
      return ExplanationMessage.WiredConnection;
    }
    return ExplanationMessage.NotConnected;
  };

  useEffect(() => {
    if (
      connectedNetwork &&
      previousConnectedNetwork?.bssid !== connectedNetwork?.bssid
    ) {
      setSelectedNetwork(connectedNetwork);
    }
  }, [connectedNetwork, previousConnectedNetwork]);

  return (
    <>
      <Layout
        banner={{
          src: wifiPageScreen,
          alt: "wifi-page-banner",
        }}
        prompt={
          <>
            Connect me to <span className="green">WiFi</span>
          </>
        }
        explanation={getExplanation()}
        nextButton={{
          onClick: onNextClick,
          disabled: !isConnected,
        }}
        skipButton={{
          label: skipButtonLabel,
          onClick: () => {
            if (!isConnected && showSkipWarning) {
              setIsSkipWarningDialogActive(true);
            } else if (onSkipClick) {
              onSkipClick();
            }
          },
        }}
        backButton={{ onClick: onBackClick }}
        showBack={!!onBackClick}
        showNext={!!onNextClick}
        showSkip={!!onSkipClick}
      >
        <div className={styles.wifiSelectContainer}>
          <Select
            // force rerender when selected ssid changes
            key={selectedBSSID}
            value={
              selectedBSSID &&
              selectedSSID && {
                value: selectedBSSID,
                label: selectedSSID,
              }
            }
            options={networks.map(({ ssid, bssid }) => ({
              value: bssid,
              label: ssid,
            }))}
            onChange={(newBSSID) => {
              setIsConnectDialogActive(true);
              setSelectedNetwork(
                networks.find((network) => newBSSID === network.bssid)
              );
            }}
            placeholder="Please select WiFi network..."
            isDisabled={isFetchingNetworks}
          />

          <div className={styles.selectDetailsContainer}>
            {isFetchingNetworks ? (
              <>
                <Spinner size={20} />{" "}
                <span className={styles.settingMessage}>
                  fetching networks...
                </span>
              </>
            ) : (
              <Button
                unstyled
                onClick={onRefreshClick}
                className={styles.refreshButton}
              >
                <img src={refreshIcon} alt="refresh-button" />
              </Button>
            )}
          </div>
        </div>

        {errorMessage && <span className={styles.error}>{errorMessage}</span>}
      </Layout>

      <ConnectDialogContainer
        active={isConnectDialogActive}
        network={selectedNetwork}
        setConnectedNetwork={setConnectedNetwork}
        onCancel={() => {
          setIsConnectDialogActive(false);
          setSelectedNetwork(undefined);
        }}
        onDone={() => {
          setIsConnectDialogActive(false);
          if (isConnected && onNextClick) {
            onNextClick();
          }
        }}
      />
      <SkipWarningDialog
        active={isSkipWarningDialogActive}
        onConnectClick={() => setIsSkipWarningDialogActive(false)}
        onSkipClick={onSkipClick || (() => {})}
      />
    </>
  );
};
