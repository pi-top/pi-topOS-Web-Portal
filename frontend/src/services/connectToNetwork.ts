import axios from "axios";

import apiBaseUrl from "./apiBaseUrl";

import { NetworkCredentials } from "../types/Network";


export default async function connectToNetwork(networkCredentials: NetworkCredentials, timeout?: number) {
  await axios.post(
    `${apiBaseUrl}/wifi-credentials`,
    networkCredentials,
    { timeout: timeout || 10000 }
  );
}
