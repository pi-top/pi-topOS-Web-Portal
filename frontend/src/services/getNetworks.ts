import api from "./api";

import { Network } from "../types/Network";

export default async function getNetworks() {
  const { data } = await api.get<Network[]>(`/wifi-ssids`);

  return data;
}
