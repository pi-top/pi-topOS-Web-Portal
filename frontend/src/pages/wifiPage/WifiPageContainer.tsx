import React, { useState, useEffect } from "react";

import WifiPage from "./WifiPage";

import getNetworks from "../../services/getNetworks";
import isConnectedToNetwork from "../../services/isConnectedToNetwork";

import { Network } from "../../types/Network";
import startVncWpaGui from "../../services/startVncWpaGui";
import getVncWpaGuiUrl from "../../services/getVncWpaGuiUrl";
import stopVncWpaGui from "../../services/stopVncWpaGui";

export type Props = {
  goToNextPage: (isConnected: boolean) => void;
  goToPreviousPage: () => void;
  connectedNetwork?: Network;
  setConnectedNetwork: (network: Network) => void;
};

export default ({
  goToNextPage,
  goToPreviousPage,
  connectedNetwork,
  setConnectedNetwork,
}: Props) => {
  const [isFetchingNetworks, setIsFetchingNetworks] = useState(false);
  const [fetchNetworksError, setFetchNetworksError] = useState(false);
  const [advancedConfigError, setAdvancedConfigError] = useState(false);
  const [networks, setNetworks] = useState<Network[]>([]);
  const [isConnected, setIsConnected] = useState(false);
  const [advancedConfigUrl, setAdvancedConfigUrl] = useState("");


  const fetchNetworks = () => {
    setIsFetchingNetworks(true);

    getNetworks()
      .then((getNetworksResponse) => setNetworks(getNetworksResponse))
      .catch(() => setFetchNetworksError(true))
      .finally(() => setIsFetchingNetworks(false));
  };

  const internetConnectionStatus = () => {
    isConnectedToNetwork()
      .then((isConnectedToNetworkResponse) =>
        setIsConnected(isConnectedToNetworkResponse.connected)
      )
      .catch(() => setIsConnected(false));
  };

  const waitForAdvancedConfigUrlTimeout = 700;

  const waitForAdvancedConfigUrl = () => {
    const interval = setInterval(async () => {
      try {
        getVncWpaGuiUrl()
          .then((data: any) => {
            if (data.url !== "") {
              clearInterval(interval)
              setAdvancedConfigUrl(data.url);
            }
          })
      } catch (_) {}
    }, waitForAdvancedConfigUrlTimeout);
  }

  const startAdvancedWifiConfig = () => {
    startVncWpaGui()
      .then(() => setTimeout(waitForAdvancedConfigUrl, 300))
      .catch(() => setAdvancedConfigError(true))
  };

  const stopAdvancedWifiConfig = () => {
    setAdvancedConfigUrl("")
    stopVncWpaGui()
      .catch(() => setAdvancedConfigError(true))
  };

  useEffect(() => {
    Promise.all([fetchNetworks(), internetConnectionStatus()]);
  }, []);

  useEffect(() => {
    if (connectedNetwork)
    {
      setIsConnected(true);
    }
  }, [connectedNetwork])

  return (
    <WifiPage
      onNextClick={() => goToNextPage(isConnected)}
      onSkipClick={() => goToNextPage(isConnected)}
      onBackClick={goToPreviousPage}
      onRefreshClick={fetchNetworks}
      onAdvancedConfigurationDialogOpen={startAdvancedWifiConfig}
      onAdvancedConfigurationDialogClose={stopAdvancedWifiConfig}
      networks={networks}
      isFetchingNetworks={isFetchingNetworks}
      isConnected={isConnected}
      connectedNetwork={connectedNetwork}
      setConnectedNetwork={setConnectedNetwork}
      fetchNetworksError={fetchNetworksError}
      advancedConfigError={advancedConfigError}
      advancedConfigUrl={advancedConfigUrl}
    />
  );
};
