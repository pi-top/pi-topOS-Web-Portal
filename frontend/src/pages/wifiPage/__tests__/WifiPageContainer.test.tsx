import React, { ReactNode } from "react";
import {
  render,
  BoundFunction,
  QueryByText,
  GetByText,
  wait,
  fireEvent,
  GetByBoundAttribute,
  AllByBoundAttribute,
  QueryByBoundAttribute,
  waitForElement,
  AllByText,
  within,
} from "@testing-library/react";
import ReactDom from "react-dom";

import WifiPageContainer, { Props } from "../WifiPageContainer";
import { ErrorMessage, ExplanationMessage } from "../WifiPage";

import { Network, NetworkCredentials } from "../../../types/Network";
import queryReactSelect from "../../../../test/helpers/queryReactSelect";
import querySpinner from "../../../../test/helpers/querySpinner";
import { KeyCode } from "../../../../test/types/Keys";
import getNetworks from "../../../services/getNetworks";
import isConnectedToNetwork from "../../../services/isConnectedToNetwork";
import connectToNetwork from "../../../services/connectToNetwork";

jest.mock("../../../services/getNetworks");
jest.mock("../../../services/isConnectedToNetwork");
jest.mock("../../../services/connectToNetwork");

const getNetworksMock = getNetworks as jest.Mock;
const isConnectedToNetworkMock = isConnectedToNetwork as jest.Mock;
const connectToNetworkMock = connectToNetwork as jest.Mock;

const originalCreatePortal = ReactDom.createPortal;

describe("WifiPageContainer", () => {
  let defaultProps: Props;
  let wifiPageContainer: HTMLElement;
  let queryByText: BoundFunction<QueryByText>;
  let getByTestId: BoundFunction<GetByBoundAttribute>;
  let getByLabelText: BoundFunction<GetByBoundAttribute>;
  let getByText: BoundFunction<GetByText>;
  let getAllByText: BoundFunction<AllByText>;
  let getByAltText: BoundFunction<GetByBoundAttribute>;
  beforeEach(async () => {
    getNetworksMock.mockResolvedValue([]);
    isConnectedToNetworkMock.mockResolvedValue({ connected: false });
    connectToNetworkMock.mockImplementation(
      (creds: NetworkCredentials) =>
        new Promise((res, rej) => {
          if (
            creds.ssid === "unsecured-ssid" ||
            creds.password === "correct-password"
          ) {
            return res();
          }

          rej();
        })
    );

    ReactDom.createPortal = jest.fn();
    const createPortalMock = ReactDom.createPortal as jest.Mock;
    createPortalMock.mockImplementation((element: ReactNode) => element);

    defaultProps = {
      goToNextPage: jest.fn(),
      goToPreviousPage: jest.fn(),
      connectedNetwork: undefined,
      setConnectedNetwork: jest.fn(),
    };
  });
  afterEach(() => {
    getNetworksMock.mockRestore();
    isConnectedToNetworkMock.mockRestore();
    connectToNetworkMock.mockRestore();
    ReactDom.createPortal = originalCreatePortal;
  });

  it("disables the next button while loading", async () => {
    ({ getByText } = render(<WifiPageContainer {...defaultProps} />));

    expect(getByText("Next")).toBeDisabled();

    await wait();
  });

  it("renders spinner while loading", async () => {
    ({ container: wifiPageContainer } = render(
      <WifiPageContainer {...defaultProps} />
    ));

    expect(querySpinner(wifiPageContainer)).toBeInTheDocument();

    await wait();
  });

  it("renders correct message while loading", async () => {
    ({ queryByText } = render(<WifiPageContainer {...defaultProps} />));

    expect(queryByText("fetching networks...")).toBeInTheDocument();

    await wait();
  });

  it("renders skip warning dialog when skip button pressed", () => {
    ({ getAllByText, getByTestId } = render(
      <WifiPageContainer {...defaultProps} />
    ));

    fireEvent.click(getAllByText("Skip")[0]);
    expect(getByTestId("skip-warning-dialog")).not.toHaveClass("hidden");
  });

  it('hides skip warning dialog when "Connect" button clicked', () => {
    ({ getAllByText, getByTestId } = render(
      <WifiPageContainer {...defaultProps} />
    ));

    fireEvent.click(getAllByText("Skip")[0]);
    expect(getByTestId("skip-warning-dialog")).not.toHaveClass("hidden");

    fireEvent.click(getByText("Connect"));
    expect(getByTestId("skip-warning-dialog")).toHaveClass("hidden");
  });

  it("skips to next page when warning dialog skip button clicked", () => {
    ({ getByTestId } = render(<WifiPageContainer {...defaultProps} />));

    fireEvent.click(
      within(getByTestId("skip-warning-dialog")).getByText("Skip")
    );

    expect(defaultProps.goToNextPage).toHaveBeenCalled();
  });

  describe("when connected network is passed", () => {
    const network = {
      ssid: "password-protected-ssid",
      passwordRequired: true,
    };

    beforeEach(() => {
      defaultProps = {
        ...defaultProps,
        connectedNetwork: network,
      };
    });

    it("renders correct explanation", async () => {
      ({
        container: wifiPageContainer,
        getByText,
        queryByText,
      } = render(<WifiPageContainer {...defaultProps} />));
      await wait();
      expect(
        queryByText(ExplanationMessage.WiFiConnection)
      ).toBeInTheDocument();
    });
  });

  describe("when is connected to network", () => {
    beforeEach(async () => {
      isConnectedToNetworkMock.mockResolvedValue({ connected: true });
      ({
        container: wifiPageContainer,
        getByText,
        queryByText,
      } = render(<WifiPageContainer {...defaultProps} />));

      await wait();
    });

    it("renders correct explanation", () => {
      expect(
        queryByText(ExplanationMessage.WiredConnection)
      ).toBeInTheDocument();
    });
  });

  describe("when not connected to network", () => {
    beforeEach(async () => {
      isConnectedToNetworkMock.mockResolvedValue({ connected: false });
      ({
        container: wifiPageContainer,
        queryByText,
        getByText,
      } = render(<WifiPageContainer {...defaultProps} />));

      await wait();
    });

    it("renders correct explanation", () => {
      expect(queryByText(ExplanationMessage.NotConnected)).toBeInTheDocument();
    });
  });

  describe("when unable to determine if connected to network", () => {
    beforeEach(async () => {
      isConnectedToNetworkMock.mockRejectedValue(
        new Error("unable to determine connection status")
      );

      ({
        container: wifiPageContainer,
        queryByText,
        getByText,
      } = render(<WifiPageContainer {...defaultProps} />));

      await wait();
    });

    it("renders correct explanation", () => {
      expect(queryByText(ExplanationMessage.NotConnected)).toBeInTheDocument();
    });
  });

  describe("when unable to get networks", () => {
    beforeEach(async () => {
      getNetworksMock.mockRejectedValue(new Error("unable to get networks"));

      ({
        container: wifiPageContainer,
        queryByText,
        getByText,
        getByAltText,
        getByTestId,
        getAllByText,
      } = render(<WifiPageContainer {...defaultProps} />));

      await wait();
    });

    it("renders error message", () => {
      expect(queryByText(ErrorMessage.FetchNetworks)).toBeInTheDocument();
    });

    describe("on refreshing networks", () => {
      let networks: Network[];
      beforeEach(() => {
        networks = [
          {
            ssid: "network-found-on-refresh-ssid",
            passwordRequired: true,
          },
        ];

        getNetworksMock.mockResolvedValueOnce(networks);
      });

      it("renders correct loading message", async () => {
        fireEvent.click(getByAltText("refresh-button"));

        expect(queryByText("fetching networks...")).toBeInTheDocument();

        await wait();
      });

      it("renders correct message while loading", async () => {
        fireEvent.click(getByAltText("refresh-button"));

        expect(queryByText("fetching networks...")).toBeInTheDocument();

        await wait();
      });

      it("renders new select options when loaded", async () => {
        fireEvent.click(getByAltText("refresh-button"));

        await wait();

        fireEvent.keyDown(queryReactSelect(wifiPageContainer)!, {
          keyCode: KeyCode.DownArrow,
        });

        networks.forEach(({ ssid }) => {
          expect(queryByText(ssid)).toBeInTheDocument();
        });
      });
    });
  });

  describe("when networks are loaded successfully", () => {
    let networks: Network[];

    beforeEach(async () => {
      networks = [
        {
          ssid: "password-protected-ssid",
          passwordRequired: true,
        },
        {
          ssid: "unsecured-ssid",
          passwordRequired: false,
        },
      ];

      getNetworksMock.mockResolvedValue(networks);

      ({
        container: wifiPageContainer,
        queryByText,
        getByText,
        getByTestId,
        getByLabelText,
      } = render(<WifiPageContainer {...defaultProps} />));

      await wait();
    });

    it("renders correct select options", () => {
      fireEvent.keyDown(queryReactSelect(wifiPageContainer)!, {
        keyCode: KeyCode.DownArrow,
      });

      networks.forEach(({ ssid }) => {
        expect(queryByText(ssid)).toBeInTheDocument();
      });
    });

    describe("when option with password is selected", () => {
      let network: Network;
      beforeEach(() => {
        // open the select
        fireEvent.keyDown(queryReactSelect(wifiPageContainer)!, {
          keyCode: KeyCode.DownArrow,
        });

        network = networks.find(({ passwordRequired }) => passwordRequired)!;

        // click the wifi option
        fireEvent.click(getByText(network.ssid));
      });

      it("shows the connect dialog", () => {
        expect(getByTestId("connect-dialog")).not.toHaveClass("hidden");
      });

      it("renders dialog message correctly", () => {
        expect(
          within(getByTestId("connect-dialog")).queryByTestId("dialog-message")
        ).toMatchSnapshot();
      });

      it("renders error when incorrect password is used", async () => {
        const passwordInputLabel = getByLabelText("Enter password below");
        fireEvent.change(passwordInputLabel, {
          target: { value: "incorrect-password" },
        });
        fireEvent.click(getByText("Join"));
        await waitForElement(() =>
          getByText(
            `There was an error connecting to ${network.ssid}... please check your password and try again`
          )
        );
      });

      it("clears error when cancel is clicked", async () => {
        const passwordInputLabel = getByLabelText("Enter password below");
        fireEvent.change(passwordInputLabel, {
          target: { value: "incorrect-password" },
        });
        fireEvent.click(getByText("Join"));
        await waitForElement(() =>
          getByText(
            `There was an error connecting to ${network.ssid}... please check your password and try again`
          )
        );
        fireEvent.click(getByText("Cancel"));
        fireEvent.keyDown(queryReactSelect(wifiPageContainer)!, {
          keyCode: KeyCode.DownArrow,
        });
        fireEvent.click(getByText(network.ssid));

        expect(
          queryByText(
            `There was an error connecting to ${network.ssid}... please check your password and try again`
          )
        ).not.toBeInTheDocument();
      });

      it("clears error when retry is clicked", async () => {
        const passwordInputLabel = getByLabelText("Enter password below");
        fireEvent.change(passwordInputLabel, {
          target: { value: "incorrect-password" },
        });
        fireEvent.click(getByText("Join"));
        await waitForElement(() =>
          getByText(
            `There was an error connecting to ${network.ssid}... please check your password and try again`
          )
        );
        fireEvent.click(getByText("Retry"));

        expect(
          queryByText(
            `There was an error connecting to ${network.ssid}... please check your password and try again`
          )
        ).not.toBeInTheDocument();
        waitForElement(() =>
          getByText(
            `There was an error connecting to ${network.ssid}... please check your password and try again`
          )
        );
      });

      describe("on cancel click", () => {
        beforeEach(() => {
          fireEvent.click(getByText("Cancel"));
        });

        it("hides the connect dialog on cancel click", () => {
          expect(getByTestId("connect-dialog")).toHaveClass("hidden");
        });

        it("resets selected network", () => {
          expect(queryByText(network.ssid)).not.toBeInTheDocument();
        });
      });
    });
  });
});
