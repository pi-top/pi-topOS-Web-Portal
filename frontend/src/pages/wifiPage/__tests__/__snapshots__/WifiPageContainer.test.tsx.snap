// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WifiPageContainer when networks are loaded successfully when option with password is selected renders dialog message correctly 1`] = `
<h3
  class="messageContainer"
  data-testid="dialog-message"
>
  The WiFi network
  <span
    class="green"
  >
    password-protected-ssid
  </span>

  requires
   a password
</h3>
`;
